.pre-commit-config.yaml

```diff
@@ -27,3 +27,14 @@ repos:
       - id: flake8
         additional_dependencies: [pydocstyle>5.1.0, flake8-docstrings>1.4.0]
         args: ["--docstring-convention=all"]
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.1.0
+    hooks:
+      - id: codespell
+        files: ^.*\.(py|c|h|md|rst|yml)$
```

Comment:
Did you try `.cpp` and `*.m` for macos as well? https://github.com/matplotlib/matplotlib/blob/main/src/_macosx.m

It looks like the extern directory is ignored, so hopefully this would only catch the cpp/objective-c that someone on the project has added.

Reply:
I had just selected the files that got caught when I ran `codespell` over all files, but I've added `.cpp` and `.m` into the extensions to run over to be future safe in the push I just did with the rebase. Also, yes, `extern/` is ignored for all of pre-commit and so safe from getting hit by this hook.